### ISSUE TITLE
Compilation Warning/Errors

### ISSUE THREAD

**Issue by peterbangert** [strymon-system/ds2#9]
Im trying to compile and run DS2, encountering some issues

system: Ubuntu 20.04
rustc --version: rustc 1.64.0 (a55dd71d5 2022-09-19)

Some of the warnings look ignorable, but the errors look like they are due to upstream version syntax changes causing issues. 

`$ cargo run --release --bin manager` output:

```
warning: value assigned to `true_processing_rate` is never read
   --> src/policy/scaling.rs:320:17
    |
320 |         let mut true_processing_rate = Rate::default();
    |                 ^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_assignments)]` on by default
    = help: maybe it is overwritten before being read?

warning: value assigned to `estimated_true_output_rate` is never read
   --> src/policy/scaling.rs:328:17
    |
328 |         let mut estimated_true_output_rate = Rate::default();
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: maybe it is overwritten before being read?

warning: value assigned to `observed_output_rate` is never read
   --> src/policy/scaling.rs:329:17
    |
329 |         let mut observed_output_rate = Rate::default();    // Used only in estimating source rate ratios
    |                 ^^^^^^^^^^^^^^^^^^^^
    |
    = help: maybe it is overwritten before being read?

warning: unused variable: `threshold`
   --> src/policy/scaling.rs:293:62
    |
293 | pub fn evaluate_scaling_policy_at_epoch(topo: &mut Topology, threshold: f64, factor: f64, epoch: Epoch, verbose: bool) -> String
    |                                                              ^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_threshold`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: variable does not need to be mutable
   --> src/dataflow/parse.rs:161:16
    |
161 |         for (_,mut logs) in instance_logs.iter_mut()
    |                ----^^^^
    |                |
    |                help: remove this `mut`
    |
    = note: `#[warn(unused_mut)]` on by default

warning: variable does not need to be mutable
   --> src/dataflow/parse.rs:209:24
    |
209 |                 for (_,mut logs) in instance_logs.iter_mut()
    |                        ----^^^^
    |                        |
    |                        help: remove this `mut`

warning: variable does not need to be mutable
   --> src/dataflow/parse.rs:505:16
    |
505 |         for (_,mut logs) in instance_logs.iter_mut()
    |                ----^^^^
    |                |
    |                help: remove this `mut`

warning: variable does not need to be mutable
   --> src/policy/scaling.rs:345:21
    |
345 |         if let Some(mut input_rate) = true_output_rate.remove(&idx)
    |                     ----^^^^^^^^^^
    |                     |
    |                     help: remove this `mut`

warning: `ds2` (lib) generated 8 warnings
   Compiling ds2 v0.1.0 (/home/pbangert/Code/msc_thesis/ds2/controller)
warning: use of deprecated macro `try`: use the `?` operator instead
   --> src/bin/manager.rs:163:5
    |
163 |     try!(watcher.watch(metrics_repo_path.as_path(), RecursiveMode::Recursive));
    |     ^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: use of deprecated macro `try`: use the `?` operator instead
   --> src/bin/manager.rs:161:43
    |
161 |     let mut watcher: RecommendedWatcher = try!(Watcher::new(tx.clone(), Duration::from_secs(policy_interval as u64)));
    |                                           ^^^

error[E0624]: associated function `new` is private
  --> src/bin/manager.rs:63:28
   |
63 |     let mut conf = Config::new();
   |                            ^^^ private associated function
   |
  ::: /home/pbangert/.cargo/registry/src/github.com-1ecc6299db9ec823/config-0.13.2/src/config.rs:39:5
   |
39 |     pub(crate) fn new(value: Value) -> Self {
   |     --------------------------------------- private associated function defined here

error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> src/bin/manager.rs:63:20
   |
63 |     let mut conf = Config::new();
   |                    ^^^^^^^^^^^-- an argument of type `Value` is missing
   |
note: associated function defined here
  --> /home/pbangert/.cargo/registry/src/github.com-1ecc6299db9ec823/config-0.13.2/src/config.rs:39:19
   |
39 |     pub(crate) fn new(value: Value) -> Self {
   |                   ^^^
help: provide the argument
   |
63 |     let mut conf = Config::new(/* Value */);
   |                    ~~~~~~~~~~~~~~~~~~~~~~~~

warning: use of deprecated associated function `config::Config::merge`: please use 'ConfigBuilder' instead
  --> src/bin/manager.rs:64:16
   |
64 |     match conf.merge(File::with_name("config/ds2.toml")) {
   |                ^^^^^

Some errors have detailed explanations: E0061, E0624.
For more information about an error, try `rustc --explain E0061`.
warning: `ds2` (bin "manager") generated 3 warnings
error: could not compile `ds2` due to 2 previous errors; 3 warnings emitted


```

**Comment by peterbangert** [comment]
seems like Config has been replaced with ConfigBuilder

**Comment by peterbangert** [comment]
i managed to get things running, and im unsure what exactly it is i did. Im using latest version of rust.

so in that regard ill close this ticket as being resolved.

**Comment by flint-stone** [comment]
@peterbangert  Could you share how did you manage to fix the problem? I'm getting the following issue while building:

```

error[E0624]: associated function `new` is private
  --> src/bin/manager.rs:63:28
   |
63 |     let mut conf = Config::new();
   |                            ^^^ private associated function
   |
  ::: /Users/lexu/.cargo/registry/src/github.com-1ecc6299db9ec823/config-0.13.2/src/config.rs:39:5
   |
39 |     pub(crate) fn new(value: Value) -> Self {
   |     --------------------------------------- private associated function defined here

error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> src/bin/manager.rs:63:20
   |
63 |     let mut conf = Config::new();
   |                    ^^^^^^^^^^^-- an argument of type `Value` is missing
   |
note: associated function defined here
  --> /Users/lexu/.cargo/registry/src/github.com-1ecc6299db9ec823/config-0.13.2/src/config.rs:39:19
   |
39 |     pub(crate) fn new(value: Value) -> Self {
   |                   ^^^
help: provide the argument
   |
63 |     let mut conf = Config::new(/* Value */);
   |                               ~~~~~~~~~~~~~

warning: use of deprecated associated function `config::Config::merge`: please use 'ConfigBuilder' instead
  --> src/bin/manager.rs:64:16
   |
64 |     match conf.merge(File::with_name("config/ds2.toml")) {
   |           
```     ^^^^^

**Comment by antiguru** [comment]
Fixed by

**Comment by antiguru** [comment]
Fixed by pinning versions in e3decca4197cff38dafa2cf50a0d2767ec08f728.
